lib/Bridge/TolerantParser/WorseReflection/DoctrineAnnotationCompletor.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Phpactor\Completion\Bridge\TolerantParser\WorseReflection;
+
+use Generator;
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\SourceFileNode;
+use Microsoft\PhpParser\Parser;
+use Phpactor\Completion\Core\Completor;
+use Phpactor\Completion\Core\Completor\NameSearcherCompletor;
+use Phpactor\Completion\Core\Suggestion;
+use Phpactor\Completion\Core\Util\OffsetHelper;
+use Phpactor\ReferenceFinder\NameSearcher;
+use Phpactor\ReferenceFinder\Search\NameSearchResult;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\TextDocument;
+use Phpactor\WorseReflection\Reflector;
+
+class DoctrineAnnotationCompletor extends NameSearcherCompletor implements Completor
+{
+    /**
+     * @var Reflector
+     */
+    private $reflector;
+
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    public function __construct(
+        NameSearcher $nameSearcher,
+        Reflector $reflector,
+        Parser $parser = null
+    ) {
+        parent::__construct($nameSearcher);
+
+        $this->reflector = $reflector;
+        $this->parser = $parser ?: new Parser();
+    }
+
+    public function complete(TextDocument $source, ByteOffset $byteOffset): Generator
+    {
+        $truncatedSource = $this->truncateSource((string) $source, $byteOffset->toInt());
+        $sourceNodeFile = $this->parser->parseSourceFile((string) $source);
+
+        $node = $this->findNodeForPhpdocAtPosition(
+            $sourceNodeFile,
+            // the parser requires the byte offset, not the char offset
+            strlen($truncatedSource)
+        );
+
+        if (!$node) {
+            // Ignore this case is the cursor is not in a phpdoc block
+            return true;
+        }
+
+        if (!$annotation = $this->extractAnnotation($truncatedSource)) {
+            // Ignore if not an annotation
+            return true;
```

Comment:
Not related, but should probably document the return type here https://github.com/elythyr/completion/blob/6bc7317089350c83a00771ba4d725b5d0ec2c944/lib/Core/Completor.php#L12

Reply:
I thought it would still return a generator, apparently not...
I will check others completor to be sure the same mistake has  not be done.
I will fix it with:
```suggestion
            yield from [];

            return true;
        
```

Which is the best solution I got so far.

Reply:
Sorry, the code is fine AFAICS. But we should document the generic type in the docblock of the interface:

```
Generator<TKey, TValue, TSend, TReturn>
```

No worries to do it here, just an omission from the existing code.

Reply:
Oups, deleted my comment. Hoping you didn't saw it ;)
I will update the doc, what is `TSend` here ?

Reply:
Send would be `null` here I suppose - send is the value you send into a generator `$generator->send('foo');` `$foo = yield;`